microsetta_interface/implementation.py

```diff
@@ -1673,13 +2092,37 @@ def get_ajax_check_activation_code(code, email):
 # surveys added to this source AFTER these samples are claimed will NOT be
 # associated with these samples.  This behavior is by design.
 @prerequisite([SOURCE_PREREQS_MET])
-def post_claim_samples(*, account_id=None, source_id=None, body=None):
+def post_claim_samples(*, account_id=None, source_id=None, body=None,
+                       sample_ids=None):
+
+    if sample_ids is not None:
```

Comment:
If it's coming directly from the form where the user selects samples, it's in the body. However, if the user attempts to claim samples before they've agreed to the biospecimen consent, we redirect them to the consent, which then returns the sample_ids as a parameter rather than in the body.

Reply:
Ah, thanks!